scripts/gateway-pre-start.sh

```
@@ -24,7 +24,22 @@ CLAWBOX_HOME_DIR="${CLAWBOX_HOME_DIR:-${HOME:-/home/clawbox}}"
+# OPENCLAW_CONFIG_PATH / OPENCLAW_STATE_DIR when set and otherwise derives
+# its tree from OPENCLAW_HOME — which it reads as the ACCOUNT home, so the
+# state lands in `$OPENCLAW_HOME/.openclaw`. ClawBox has always used that
+# name for the .openclaw directory itself, and the updater exported it into
+# this script: the CLI then built a second tree at ~/.openclaw/.openclaw/ and
+# wrote the memory-search switch there while the real config stayed
+# half-written (2026-09-04). The two canonical overrides win over
+# OPENCLAW_HOME, and the misread name is dropped from the environment so no
+# child can inherit it. Neither reaches the gateway: ExecStartPre's
+# environment ends with it.
+export OPENCLAW_CONFIG_PATH="$OPENCLAW_CONFIG"
+export OPENCLAW_STATE_DIR="$(dirname "$OPENCLAW_CONFIG")"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)